src/cogent3/core/new_alignment.py

```diff
@@ -356,35 +395,72 @@ def take_seqs(
         negate
             select all sequences EXCEPT names
         kwargs
-            keyword arguments to be passed to the constructor of the new seqcollection
+            keyword arguments to be passed to the constructor of the new collection
+
+        Notes
+        -----
+        The seqs in the new collection will be references to the same objects as
+        the seqs in the old collection.
         """
+        # todo: kath, note gavins comment that this method could operate on only the names
+        # list and not the seqs_data object
 
-        # todo: kath, add subset method onto SeqsData
         if negate:
-            selected_seqs = {
-                name: self.seqs.get_seq_str(seqid=name)
-                for name in self.names
-                if name not in names
-            }
+            names = [name for name in self.names if name not in names]
         else:
-            selected_seqs = {
-                name: self.seqs.get_seq_str(seqid=name)
-                for name in names
-                if name in self.names
-            }
+            names = [name for name in names if name in self.names]
 
-        if not selected_seqs:
+        if not names:
             return {}
 
-        moltype = kwargs.pop("moltype", self.moltype)
+        seqs_data = self.seqs.subset(names)
 
-        seqs_data = SeqsData(data=selected_seqs, alphabet=moltype.alphabet)
+        moltype = kwargs.pop("moltype", self.moltype)
         result = self.__class__(
             seqs_data=seqs_data, moltype=moltype, info=self.info, **kwargs
         )
-        # todo: kath, update annotation_db and assign to result when we have annotations working
+        if self.annotation_db:
+            result.annotation_db = type(self.annotation_db)()
+            result.annotation_db.update(
+                annot_db=self.annotation_db, seqids=result.names
+            )
         return result
 
+    def get_seq_indices(
+        self, f: Callable[[new_seq.Sequence], bool], negate: bool = False
+    ):
+        """Returns list of names of seqs where f(seq) is True."""
+        get = self.seqs.__getitem__
+
+        if negate:
+
+            def new_f(x):
```

Comment:
I think this use case is likely common enough that we should add a class to `cogent3.util.misc` like the following and apply it in all those cases. The main reason to avoid such local functions is they cause problems for pickling (required for parallel computations)
```python
import typing

class negate_condition:
    """negates the result of calling func."""
    def __init__(self, func: typing.Callable[[typing.Any], bool]) -> None:
        self.func = func
    
    def __call__(self, *args: typing.Any, **kwargs: typing.Any) -> bool:
        return not bool(self.func(*args, **kwargs))
```